src/test/kotlin/com/cosmotech/connector/utils/AzureDigitalTwinsUtilTest.kt

```diff
@@ -208,47 +208,82 @@ class AzureDigitalTwinsUtilTest: AbstractUnitTest() {
-        )
-        val mainDTProperties = mutableMapOf(
-            "OutputType" to "string"
-        )
-        val extensionDT = DTDLModelInformation(
-            "dtmi:com:cosmotech:supply:TransportOperation;1",
-            true,
-            listOf("dtmi:com:cosmotech:supply:Operation;1"),
-            extensionProperties,
-            dtWithExtendDTDLModel
+    fun test_retrievePropertiesFromBaseModels() {
+        val actionDT = DTDLModelInformation(
+            "dtmi:com:cosmotech:supply:Action;1",
+            null,
+            mutableMapOf(
+                "Date" to "string",
+                "Priority" to "integer"
+            ),
+            ""
         )
-        val mainDT = DTDLModelInformation(
+        val opDT = DTDLModelInformation(
             "dtmi:com:cosmotech:supply:Operation;1",
-            false,
+            listOf("dtmi:com:cosmotech:supply:Action;1"),
+            mutableMapOf(
+                "Team" to "string"
+            ),
+            ""
+        )
+        val transportDT = DTDLModelInformation(
+            "dtmi:com:cosmotech:supply:Transport;1",
             null,
-            mainDTProperties,
-            dtWithoutExtendDTDLModel
+            mutableMapOf(
+                "Duration" to "integer",
+                "TransportType" to "enum"
+            ),
+            ""
+        )
+        val transportOpDT = DTDLModelInformation(
+            "dtmi:com:cosmotech:supply:TransportOperation;1",
+            listOf("dtmi:com:cosmotech:supply:Operation;1", "dtmi:com:cosmotech:supply:Transport;1"),
+            mutableMapOf(
+                "Capacity" to "integer"
+            ),
+            ""
         )
         val modelInformationList = mutableListOf(
-            extensionDT,
-            mainDT
+            actionDT, opDT, transportDT, transportOpDT
         )
 
-        assertFalse(extensionDT.properties.keys.containsAll(mainDTProperties.keys))
+        println("before")
+        print(actionDT)
+        print(opDT)
+        print(transportDT)
+        println(transportOpDT)
         val completedModelInformationList =
-            AzureDigitalTwinsUtil.retrievePropertiesFromExtendedModel(modelInformationList)
-        assertEquals(2,completedModelInformationList.size)
-        assertTrue(completedModelInformationList.contains(mainDT))
-        assertTrue(completedModelInformationList.contains(extensionDT))
-        assertNotNull(extensionDT.properties)
-        assertEquals(
-            4,
-            extensionDT.properties.size)
-        assertTrue(extensionDT.properties.keys.containsAll(mainDTProperties.keys))
-        for ((key,value) in mainDTProperties)
-            assertEquals(extensionDT.properties[key], value)
+            AzureDigitalTwinsUtil.retrievePropertiesFromBaseModels(modelInformationList)
+        println("after")
+        print(actionDT)
+        print(opDT)
+        print(transportDT)
+        println(transportOpDT)
```

Comment:
Could you use a logger instead of the `print*` statements ? This would allow to have more control when running the tests.
You can declare one as a private member with:
```kotlin
private val logger = org.slf4j.LoggerFactory.getLogger(AzureDigitalTwinsUtilTest::class.java)
```